### DISCUSSION TITLE
How can I make my NITRO_PRESET=custom for firebase

### DISCUSSION THREAD

**Discussion by nandordudas** [nuxt/nuxt#17879] [Questions]
Hello,

I want to make my custom preset (for firebase), but I need more information about that.

I tried to create firebase cloud function triggers (for firestore), but it doesn't work. It is normal, because the builder wraps only with `functions.https.onRequest` the handler - I can see in compiled code.

I see some discussions and issues about changing or set different cloud region. And I think the main reason is common.

Where is happening the magic?
I checked the related nitro preset, but I didn't find the part of wrapping magic - mea culpa.

My plan was that: I will create a new api file (server/api/index.ts) and here I make the hack, the hack with variable names and exports.
But now, I'm thinking about a hookish thing, because I need cloud function triggers and modifiers use beside the onRequest.

```ts
// /server/api/index.ts (doesn't work with default export either)
...
export const createPet = firestore.document(...).onCreate(...)
```

I love this feature in nuxt and I try to find a solution, but I need some help.

Thank you in advance, take care

**Comment by danielroe** [comment]
Check out https://v3.nuxtjs.org/docs/deployment/presets/custom.
